ui/perfherder/health/HealthView.jsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Col, Container, Row } from 'reactstrap';
+
+import withValidation from '../Validation';
+import { getFrameworkData } from '../helpers';
+import LoadingSpinner from '../../shared/LoadingSpinner';
+import {
+  errorMessageClass,
+  genericErrorMessage,
+} from '../../helpers/constants';
+import ErrorBoundary from '../../shared/ErrorBoundary';
+import { getData } from '../../helpers/http';
+import { createApiUrl } from '../../helpers/url';
+
+import HealthTableControls from './HealthTableControls';
+
+class HeathView extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.validated = this.props.validated;
+    this.state = {
+      framework: getFrameworkData(this.props),
+      loading: false,
+      projectsMap: false,
+      platformsMap: false,
+    };
+  }
+
+  componentDidMount() {
+    const { projects, platforms, updateAppState } = this.props;
```

Comment:
I think the code from componentDidMount related to `platforms` can be moved into a different function. It would make the coude decoupled. The function would handle, separately, `platforms` creation. Also, it would be easy testable.